### PR TITLE
EVG-6108 Add / Remove rejected outliers from whitelist.

### PR DESCRIPTION
**Pull request by jimoleary**

### CODE REVIEW

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+  $scope.getAggChain = (state) => {
+    let chain = [];
+
+    // Check if the state has filters
+    let filteringChain = MDBQueryAdaptor.compileFiltering(
+      // filters context enhances state data with important meta data
+      state.getFilteringContext()
+    );
+
+    if(!filteringChain) {
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBRejectsCtrl.js]: [nit] Everywhere else in this file, there is a space between the 'if' and the '(', could we do that here and on the next 'if' as well?

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+  //    get the remote state from atlas
+  //    renders the data (if it is the latest data load event)
+  //    clears the loading indicator.
+  // Setting filters and updating other vm state is done through relaod() and most methods should
+  // use reload..
+  function loadData() {
+    vm.isLoading = true;
+    vm.gridOptions.data = [];
+    const promises = {
+      docs:Stitch.use(STITCH_CONFIG.PERF).query((db) => {
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBRejectsCtrl.js]: [nit] Could we put a space between the ':' and the values in this object?

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js [hunk 3]
```diff
@@ -0,0 +1,385 @@
+mciModule.controller('PerfBBRejectsCtrl', function (
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBRejectsCtrl.js]: This is more of a long-term comment and I don't it expect it addressed in this PR, but I'm wondering if there is anything we can do to reduce the amount of duplication between the PerfBB controllers.

Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBRejectsCtrl.js]: I think we need to pull functionality into factories (like RejectState here). 

There is a fair amount of duplication, but there are also subtle differences.

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js (hunk 1 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBRejectsCtrl.js]: :+1:

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js (hunk 2 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBRejectsCtrl.js]: done

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js (hunk 3 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBRejectsCtrl.js]: @john-m-liu IMO no, the outliers are per test whereas you whitelist per task.

I think trying to visually represent these 2 different concepts in the same list might be difficult (although I'm not sure how well it is done at the moment anyway).

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+  const vm = this;
+  const project = window.project;
+  const LIMIT = 2000;
+
+  const updateToWhitelist = function(items, func, mark) {
+    const task_revisions = _.chain(items).
+                              map((item) => _.pick(item, 'revision', 'project', 'variant', 'task')).
+                              uniq((item) => _.values(item).join('-')).
+                              value();
+    $log.debug(task_revisions);
```
[the remark this answers is not in the document]
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBRejectsCtrl.js]: :+1:

**Pull request merged**